brave-core/src/main/java/com/github/kristofa/brave/LocalTracer.java

```diff
@@ -129,19 +132,21 @@ public SpanId startNewSpan(String component, String operation, long timestamp) {
      * Completes the span, assigning the most precise duration possible.
      */
     public void finishSpan() {
-        long endTick = System.nanoTime();
-
         Span span = spanAndEndpoint().span();
         if (span == null) return;
 
-        Long startTick = span.startTick;
-        final long duration;
-        if (startTick != null) {
-            duration = Math.max(1L, (endTick - startTick) / 1000L);
+        internalFinishSpan(span, getDuration(span));
+    }
+
+    private long getDuration(Span span) {
```

Comment:
ps do we actually need this change? timestamp is guaranteed to be set at construction time if using LocalTracer..